src/router/routes/genericResourceRoute.ts

```diff
@@ -135,23 +155,36 @@ export default class GenericResourceRoute {
                     const resourceType = req.baseUrl.substr(1);
                     const searchParamQuery = req.query;
 
-                    const allowedResourceTypes = await this.authService.getAllowedResourceTypesForOperation({
-                        operation: 'search-type',
-                        userIdentity: res.locals.userIdentity,
-                    });
+                    const updatedReadResponse = await handleSearch(res, resourceType, searchParamQuery);
+                    res.send(updatedReadResponse);
+                }),
+            );
+            this.router.post(
+                '/_search',
+                RouteHelper.wrapAsync(async (req: express.Request, res: express.Response) => {
+                    // Get the ResourceType looks like '/Patient'
+                    const resourceType = req.baseUrl.substr(1);
+                    const searchParamQuery = req.query;
+                    const { body } = req;
 
-                    const response = await this.handler.typeSearch(
-                        resourceType,
-                        searchParamQuery,
-                        allowedResourceTypes,
-                        res.locals.userIdentity,
-                    );
-                    const updatedReadResponse = await this.authService.authorizeAndFilterReadResponse({
-                        operation: 'search-type',
-                        userIdentity: res.locals.userIdentity,
-                        readResponse: response,
+                    const conflictSearchParams: string[] = [];
+                    forOwn(searchParamQuery, (value, key) => {
+                        if (has(body, key) && body[key] !== value) {
+                            conflictSearchParams.push(key);
+                        }
                     });
-                    res.send(updatedReadResponse);
+
+                    if (isEmpty(conflictSearchParams)) {
+                        const updatedReadResponse = await handleSearch(res, resourceType, {
+                            ...searchParamQuery,
+                            ...body,
+                        });
+                        res.send(updatedReadResponse);
+                    } else {
+                        throw new createError.BadRequest(
+                            `Can not process search request, query parameter and request body parameter conflicts in [${conflictSearchParams}]`,
+                        );
```

Comment:
duplicate params are actually ok.

https://www.hl7.org/fhir/http.html#search
> This has exactly the same semantics as the equivalent GET command. Note that in the POST variant, parameters may appear in both the URL and the body. Parameters have the same meaning in either place. Since parameters can repeat, putting them in both places is the same as repeating them (which is valid for some parameters and not for others).

I think it's better for the the router to not validate any search parameters. It can just forward them to the search component and it's up to search to decide if repeated parameters make sense.

Granted, this is a bit hard to test since currently fwoa crashes when search parameters are repeated...
The only search params that are properly handled when repeated are `_include` and `_revinclude` (and their `:iterate` versions)
i.e.
```
{{API_URL}}/MedicationRequest?_include=MedicationRequest:subject&_include=MedicationRequest:performer&_revinclude=MedicationAdministration:request&_revinclude=Provenance:target
```

Reply:
Thanks for the detailed explanation Nestor! Just so I understand the expected behavior, for the above example you gave, the parameters passed to the search component would be:  

```
{
  _includ: [MedicationRequest:subject, MedicationRequest:performer],
  _revinclude: [MedicationAdministration:request, Provenance:target]
}
```

And in the case where repeated parameters don't make sense (e.g. active=true&active=false), routing should still pass `active:[true, false]` for search component to decided. 

Is that the right way to understand it?

Reply:
taht's correct